### PR TITLE
Fix dropdown menu button label issue

### PR DESCRIPTION
**Pull request by huijing**
Fixes https://github.com/lorenzolewis/starlight-utils/issues/104

This PR replaces `{Astro.props.starlightProps.labels["menuButton.accessibleLabel"]}` with `{Astro.locals.t("menuButton.accessibleLabel")}`

### CODE REVIEW

**Comment by lorenzolewis** [comment]
Thank you for this! I believe we will also need to bump the minimum version of Starlight in package.json for the version in which this change occurred. Will you be able to do that as well as add a changeset to this PR? This would be a "minor" change to reflect Starlight's versioning. 

Let me know if you need any help with that and thank you!

**Comment by huijing** [comment]
I ran `npx changeset` and added the changeset, however, I'm not sure if I did the package.json bump correctly. The Starlight version with the change was 0.28.0 which requires the Astro version to be 4.14.0 (https://github.com/withastro/starlight/releases/tag/%40astrojs%2Fstarlight%400.28.0), so I changed that too. But again, not sure if that's how it's supposed to work.

File: .changeset/heavy-students-exercise.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@lorenzo_lewis/starlight-utils": minor
+---
+
+Update translation string for dropdown menu label with new Starlight localization API
```
Comment (lorenzolewis) [inline .changeset/heavy-students-exercise.md:5]:
```suggestion
Support translated strings introduced in the Starlight 0.28 localization API
```

File: packages/starlight-utils/package.json [hunk 2]
```diff
@@ -13,8 +13,8 @@
   },
   "scripts": {},
   "devDependencies": {
-    "@astrojs/starlight": "^0.25.3",
-    "astro": "^4.12.3"
+    "@astrojs/starlight": "^0.28.0",
+    "astro": "^4.14.0"
   },
   "peerDependencies": {
     "@astrojs/starlight": ">=0.16.0",
```
Comment (lorenzolewis) [inline packages/starlight-utils/package.json:20]: I believe it's the peer dependency that needs updated. The dev dependency is what is used while working on and developing THIS package, whereas the peer dependency lets us give a specific range for someone consuming THIS package. 

I think it's okay to leave the other change you've made here as it needed to be updated anyway 😁

**Review by lorenzolewis — CHANGES_REQUESTED** [review]
Thanks! A few quick changes and then it should be good to go!

File: packages/starlight-utils/package.json (hunk 2 above)
Comment (huijing) [reply packages/starlight-utils/package.json]: Ok, I made the bump.

**Review by lorenzolewis — APPROVED** [review]

**Comment by lorenzolewis** [comment]
Thanks! Just needed to change a few more things with the lock files to make it work with the site used to test.

**Pull request merged**